demos/LowAllocationWebServer/IFormatterHttpExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Text.Formatting;
+
+namespace System.Net.Http.Buffered
+{
+    public static class IFormatterHttpExtensions
+    {
+        private const string Http = "HTTP/";
+        private const int ULongMaxValueNumberOfCharacters = 20;
+
+        public static void WriteHttpStatusLine<TFormatter>(
+            this TFormatter formatter, 
+            string version, 
+            string statusCode,
+            string reasonCode) where TFormatter : IFormatter
```

Comment:
All these string will cost in encoding in the mainline scenario (when formatting UTF8/ASCII response).